nexus/src/app/sagas/instance_create.rs

```diff
@@ -1281,15 +1331,20 @@ async fn sic_instance_ensure(
     if !params.create_params.start {
         let instance_id = db_instance.id();
         // If we don't need to start the instance, we can skip the ensure
-        // and just update the instance runtime state to `Stopped`
+        // and just update the instance runtime state to `Stopped`.
+        //
+        // TODO-correctness: This is dangerous if this step is replayed, since
+        // a user can discover this instance and ask to start it in between
+        // attempts to run this step.
```

Comment:
Another approach might be to make the "Creating" DB record the output of an earlier saga step instead of refetching on line 1324. That way, if the initial record has generation 1, all the attempts to run this step will attempt to transition to generation 2, but only the first will succeed. I added a comment describing this in 7e4c45b6f.